src/main/scala/viper/silver/parser/ParseAst.scala

```diff
@@ -1224,30 +1236,23 @@ case class PAssume(e: PExp)(val pos: (Position, Position)) extends PStmt
 
 case class PInhale(e: PExp)(val pos: (Position, Position)) extends PStmt
 
-case class PVarAssign(idnuse: PIdnUse, rhs: PExp)(val pos: (Position, Position)) extends PStmt
-
-case class PFieldAssign(fieldAcc: PFieldAccess, rhs: PExp)(val pos: (Position, Position)) extends PStmt
-
-case class PMacroAssign(call: PCall, exp: PExp)(val pos: (Position, Position)) extends PStmt
+/** Can also represent a method call or statement macro with no `:=` when `targets` is empty. */
+case class PAssign(targets: Seq[PAssignTarget], rhs: PExp)(val pos: (Position, Position)) extends PStmt
 
 case class PIf(cond: PExp, thn: PSeqn, els: PSeqn)(val pos: (Position, Position)) extends PStmt
 
 case class PWhile(cond: PExp, invs: Seq[PExp], body: PSeqn)(val pos: (Position, Position)) extends PStmt
 
-case class PLocalVarDecl(idndef: PIdnDef, typ: PType, init: Option[PExp])(val pos: (Position, Position)) extends PStmt with PTypedDeclaration with PLocalDeclaration
+case class PLocalVarDecl(vars: Seq[PFormalArgDecl], init: Option[PAssign])(val pos: (Position, Position)) extends PStmt
```

Comment:
This seems odd to me: If I'm understanding this correctly, the PLocalVarDecl doesn't just contain an expression that's assigned, but instead contains a whole assign statement that explicitly enumerates all the same variables again, so it can exist independently of the declaration statement. If that's the case, why do we need the assignment to be a part of the PLocalVarDecl statement at all? We could instead have a PLocalVarDecl that only declares variables, and for declarations that contain an initialization, the parser could just return a sequence of two statements (Decl, then assignment).

Reply:
We could, but the parsing becomes rather awkward in that case. The `def stmt(implicit ctx : P[_]) : P[PStmt] =` parser expects a single statement to be returned and I'm not sure how I'd nicely adapt that to return a `(PStmt, Option[PStmt])`